scrapers/bank_of_mauritius.py

```diff
@@ -0,0 +1,81 @@
+from typing import List
+
+import requests
+from playwright.sync_api import sync_playwright
+from utils import (
+    download_directory,
+    get_browser,
+    get_file_path,
+    get_md5hash,
+    get_standardized_filename,
+    parse_arguments,
+    validate_and_get_download_link,
+)
+
+BASE_URL = "https://www.bom.mu"
+CIRCULAR_PATH = "/financial-stability/supervision/guideline"
+
+
+def get_pdf_links(base_url: str, circular_path: str, count: int = 1) -> List[str]:
+    """Scraper function for circulars published by Bank of Mauritius
+
+    Args:
+        base_url (str): Home page of bank
+        circular_path (str): URL path containing circular publications
+        count (int, optional): Number of most recent circulars desired. Defaults to 1.
+
+    Returns:
+        List[str]: The URL path of the PDF
+    """
+    with sync_playwright() as pw:
+        browser = get_browser(pw)
+        page = browser.new_page()
+        page.goto(f"{base_url}{circular_path}")
+
+        # Since the PDFs are linked in each page individually,
+        # need to first page the page link
+        page_links = page.locator("div.col-sm-8 > a").all()[0:count]
+        links = [page_link.get_attribute("href") for page_link in page_links]
+
+        pdf_links = []
+        for link in links:
+            page.goto(f"{base_url}{link}")
+            pdf_icons = page.locator('xpath=//img[contains(@src, "pdf-icon.png")]').all()
+            for icon in pdf_icons:
+                if (href := icon.locator("..").get_attribute("href")).endswith(".pdf"):
+                    pdf_links.append(href)
+
+        browser.close()
+        return pdf_links
+
+
+def bank_of_mauritius_scraper() -> None:
+    """Runner function to get PDF links and download files
+
+    Raises:
+        ValueError: If links scraped are strings
+        TypeError: If url contents are not PDFs
+    """
+    session = requests.Session()
+
+    link = get_pdf_links(BASE_URL, CIRCULAR_PATH)[0]
+
+    try:
+        absolute_url = validate_and_get_download_link(BASE_URL, link)
+    except:
+        raise
```

Comment:
This try/except is redundant here. The reason why others have it here is because they need to close the browser. Since you close it before hand you can just do

```suggestion
      absolute_url = validate_and_get_download_link(BASE_URL, link)
```

Reply:
apologies, silly me!